src/pages/Network/views/Identities.tsx

```diff
@@ -0,0 +1,193 @@
+// Copyright © 2022 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PermIdentityIcon from '@mui/icons-material/PermIdentity';
+import { Grid } from '@mui/material';
+import React, { useContext, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { FilterButton } from '../../../components/Filters/FilterButton';
+import { FilterModal } from '../../../components/Filters/FilterModal';
+import { Header } from '../../../components/Header';
+import { ChartTableHeader } from '../../../components/Headers/ChartTableHeader';
+import { HashPopover } from '../../../components/Popovers/HashPopover';
+import { FFTableText } from '../../../components/Tables/FFTableText';
+import { DataTable } from '../../../components/Tables/Table';
+import { ApplicationContext } from '../../../contexts/ApplicationContext';
+import { DateFilterContext } from '../../../contexts/DateFilterContext';
+import { FilterContext } from '../../../contexts/FilterContext';
+import { SnackbarContext } from '../../../contexts/SnackbarContext';
+import {
+  FF_Paths,
+  IDataTableRecord,
+  IdentityFilters,
+  IIdentity,
+  IPagedIdentityResponse,
+} from '../../../interfaces';
+import { DEFAULT_PADDING, DEFAULT_PAGE_LIMITS } from '../../../theme';
+import { fetchCatcher, getFFTime } from '../../../utils';
+import { hasIdentityEvent } from '../../../utils/wsEvents';
+
+export const NetworkIdentities: () => JSX.Element = () => {
+  const { newEvents, lastRefreshTime, clearNewEvents, selectedNamespace } =
+    useContext(ApplicationContext);
+  const { dateFilter } = useContext(DateFilterContext);
+  const { filterAnchor, setFilterAnchor, filterString } =
+    useContext(FilterContext);
+  const { reportFetchError } = useContext(SnackbarContext);
+  const { t } = useTranslation();
+  const [isMounted, setIsMounted] = useState(false);
+
+  const [identities, setIdentities] = useState<IIdentity[]>();
+  const [identitiesTotal, setIdentitiesTotal] = useState(0);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(DEFAULT_PAGE_LIMITS[1]);
+
+  useEffect(() => {
+    setIsMounted(true);
+    return () => {
+      setIsMounted(false);
+    };
+  }, []);
```

Comment:
I'm really not sure what this does... I suppose it's something clever, but it seems a bit magical.

Reply:
Occasionally, there are times when an async request tries to set state on a component that is not mounted anymore since the page has been navigated from. Having an `isMounted` check before each api call and state change prevents memory leaks like this